### PR TITLE
LinkButton: No wrapper API

### PR DESCRIPTION
**Pull request by somethiiing**
- Create LinkButton Component
- Implement stories + tests

The approach we went with is to use an a tag and stylize it as intended (with hover/focus/active styles). We plan on creating wrappers in Cambly-Frontend and use those as wrappers there to prevent people from using the standard A tags.

In the future, we plan on also adding a lint rule that prevents people from using this LinkButton directly (and to only use the wrapped ones).

### CODE REVIEW

File: packages/syntax-core/src/colors/foregroundColor.ts [hunk 1]
```diff
@@ -15,3 +15,20 @@ export default function foregroundColor(color: (typeof Color)[number]): string {
       return styles.whiteColor;
   }
 }
+
+export function foregroundTypographyColor(
```
Comment (christianvuerings) [inline packages/syntax-core/src/colors/foregroundColor.ts:19]: @somethiiing can we move this to its own file

File: packages/syntax-core/src/Button/ButtonConstants.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import styles from "./Button.module.css";
+
+export const textVariant = {
```
Comment (christianvuerings) [inline packages/syntax-core/src/Button/ButtonConstants.ts:3]: @somethiiing can we split this out to different files

File: packages/syntax-core/src/LinkButton/LinkButton.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+      data-testid={dataTestId}
+      className={classNames(
+        styles.linkButton,
+        buttonStyles.button,
+        foregroundColor(color),
+        backgroundColor(color),
+        buttonStyles[size],
+        {
+          [buttonStyles.fullWidth]: fullWidth,
+          [styles.fitContent]: !fullWidth,
```
Comment (christianvuerings) [inline packages/syntax-core/src/LinkButton/LinkButton.tsx:104]: @somethiiing can we see if we can remove it!

File: packages/syntax-core/src/LinkButton/LinkButton.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+  color = "primary",
+  size = "md",
+  fullWidth = false,
+  startIcon: StartIcon,
+  endIcon: EndIcon,
+}: {
+  /**
+   * Test id for the button
+   */
+  "data-testid"?: string;
```
Comment (christianvuerings) [inline packages/syntax-core/src/LinkButton/LinkButton.tsx:54]: @somethiiing can we add an onClick

File: packages/syntax-core/src/Button/Button.module.css [hunk 5]
```diff
@@ -130,23 +130,3 @@
   stroke-dashoffset: 0;
   transform-origin: center;
 }
-
-/** TODO Change with `Typography` component when available */
-.buttonText {
-  color: inherit;
-  font-family: -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", Roboto,
-    Helvetica, Arial, sans-serif;
-  font-weight: 500;
```
Comment (christianvuerings) [inline packages/syntax-core/src/Button/Button.module.css:139]: @somethiiing looks like we lost the boldness of the button text - could you work with @ajmooo to see if we can use `bold` or `semiBold` - if not, feel free to add a `span` with a classNames inside of `Typography` which has `font-weight: 500;`

File: packages/syntax-core/src/LinkButton/LinkButton.tsx [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+  /**
+   * The text to be displayed inside the button
+   */
+  text: string;
+  /**
+   * The color of the button
+   *
+   * @defaultValue "primary"
+   */
+  color?: (typeof Color)[number];
```
Comment (christianvuerings) [inline packages/syntax-core/src/LinkButton/LinkButton.tsx:64]: @somethiiing ideally we make this more specific - e.g. I know that we don't support a `gray900` `Button`, so can we only list the available options?

File: packages/syntax-core/src/LinkButton/LinkButton.tsx [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+  /**
+   * The icon to be displayed at the start of the button. Please use a [Rounded Material Icon](https://material.io/resources/icons/?style=round)
+   */
+  startIcon?: React.ComponentType<{ className: string }>;
+  /**
+   * The icon to be displayed at the end of the button. Please use a [Rounded Material Icon](https://material.io/resources/icons/?style=round)
+   */
+  endIcon?: React.ComponentType<{ className: string }>;
+  /**
+   * The custom wrapper for LinkButton, This wrapper is used to change the default "a" tag to something like Next.js Link or react-router-dom's Link.
```
Comment (christianvuerings) [inline packages/syntax-core/src/LinkButton/LinkButton.tsx:90]: @somethiiing could we update the comments which mention the wrapper?

File: packages/syntax-core/src/LinkButton/LinkButton.tsx [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+export default function LinkButton({
+  text,
+  "data-testid": dataTestId,
+  color = "primary",
+  size = "md",
+  fullWidth = false,
+  startIcon: StartIcon,
+  endIcon: EndIcon,
+}: {
+  /**
```
Comment (christianvuerings) [inline packages/syntax-core/src/LinkButton/LinkButton.tsx:51]: @somethiiing could we add `target` and `href` attributes now that we're going to return an anchor tag: `<a />`

File: packages/syntax-core/src/LinkButton/LinkButton.tsx [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+import { type Color, type Size } from "../constants";
+import Typography from "../Typography/Typography";
+
+import buttonStyles from "../Button/Button.module.css";
+import { textVariant, iconSize } from "../Button/ButtonConstants";
+
+import styles from "./LinkButton.module.css";
+
+/**
+ * LinkButton is a "variation" of Button that should look identical to Button, but should be used to render links instead.
```
Comment (christianvuerings) [inline packages/syntax-core/src/LinkButton/LinkButton.tsx:16]: @somethiiing Could you also add a link to the docs like we do now with our other components?

https://github.com/Cambly/syntax/blob/4fb87f34874c47cac6e037f5d77a4ba443e05a76/packages/syntax-core/src/TextField/TextField.tsx#L12

File: packages/syntax-core/src/LinkButton/LinkButton.tsx (hunk 3 above)
Comment (somethiiing) [reply packages/syntax-core/src/LinkButton/LinkButton.tsx:104]: checked, still needed

File: packages/syntax-core/src/LinkButton/LinkButton.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+    },
+    onClick: { action: "clicked" },
+  },
+  tags: ["autodocs"],
+} as Meta<typeof LinkButton>;
+
+export const Default: StoryObj<typeof LinkButton> = {
+  args: {
+    text: "Call to action",
+    href: "www.google.com",
```
Comment (christianvuerings) [inline packages/syntax-core/src/LinkButton/LinkButton.stories.tsx:52]: @somethiiing Seems like this links to `https://syntax-git-linkbutton-add-component-cambly.vercel.app/www.google.com`. Could we change the `www.google.com` to `https://www.cambly.com`?

File: packages/syntax-core/src/LinkButton/LinkButton.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+  component: LinkButton,
+  parameters: {
+    design: {
+      type: "figma",
+      url: "https://www.figma.com/file/p7LKna9JMU0JEkcKamzs53/Cambly-Design-System-(Draft)?node-id=994%3A2346",
+    },
+  },
+  argTypes: {
+    color: {
+      text: {
```
Comment (christianvuerings) [inline packages/syntax-core/src/LinkButton/LinkButton.stories.tsx:16]: @somethiiing why and `text`, `href` and `target` nested under `color`? Can we move them out?

File: packages/syntax-core/src/LinkButton/LinkButton.stories.tsx [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+        control: { type: "text" },
+      },
+      href: {
+        control: { type: "text" },
+      },
+      target: {
```
Comment (christianvuerings) [inline packages/syntax-core/src/LinkButton/LinkButton.stories.tsx:22]: @somethiiing Can we make `target` more restrictive? Just like we defined our types: `target?: "_blank" | "_self" | "_parent" | "_top";`

File: packages/syntax-core/src/LinkButton/LinkButton.stories.tsx [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+        control: { type: "text" },
+      },
+      options: [
```
Comment (christianvuerings) [inline packages/syntax-core/src/LinkButton/LinkButton.stories.tsx:25]: @somethiiing selecting a different color in storybook does not seem to update the color of the LinkButton

File: packages/syntax-core/src/LinkButton/LinkButton.stories.tsx [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+export const Secondary: StoryObj<typeof LinkButton> = {
+  args: { ...Default.args, color: "secondary" },
+};
+export const Tertiary: StoryObj<typeof LinkButton> = {
+  args: { ...Default.args, color: "tertiary" },
+};
+export const DestructivePrimary: StoryObj<typeof LinkButton> = {
+  args: { ...Default.args, color: "destructive-primary" },
+};
+export const DestructiveSecondary: StoryObj<typeof LinkButton> = {
```
Comment (christianvuerings) [inline packages/syntax-core/src/LinkButton/LinkButton.stories.tsx:76]: @somethiiing colors don't seem to work for these:

image

File: packages/syntax-core/src/LinkButton/LinkButton.test.tsx [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+    render(
+      <LinkButton
+        text="button"
+        data-testid="linkButton-test-id"
+        href="https://www.google.com"
+        onClick={handleClick}
+      />,
+    );
+    const linkButton = screen.getByTestId("linkButton-test-id");
+    // eslint-disable-next-line testing-library/no-await-sync-events
```
Comment (christianvuerings) [inline packages/syntax-core/src/LinkButton/LinkButton.test.tsx:68]: @somethiiing can you rebase / merge in `main` and remove this eslint disable?

We should have fixed all of these in https://github.com/Cambly/syntax/pull/185

File: packages/syntax-core/src/LinkButton/LinkButton.tsx [hunk 16]
```diff
@@ -0,0 +1,127 @@
...
+   */
+  startIcon?: React.ComponentType<{ className: string }>;
+  /**
+   * The icon to be displayed at the end of the button. Please use a [Rounded Material Icon](https://material.io/resources/icons/?style=round)
+   */
+  endIcon?: React.ComponentType<{ className: string }>;
+  /**
+   * An optional onClick event. This is used for certain wrapper's support (such as react-router-dom).
+   */
+  onClick?: React.MouseEventHandler;
```
Comment (christianvuerings) [inline packages/syntax-core/src/LinkButton/LinkButton.tsx:89]: @somethiiing Can we be more specific with the type? `React.MouseEventHandler<HTMLAnchorElement>` - just like we do with our other `onClick`s in Syntax:
https://github.com/search?q=repo%3ACambly%2Fsyntax%20%22onClick%3F%22&type=code

File: packages/syntax-core/src/LinkButton/LinkButton.tsx [hunk 17]
```diff
@@ -0,0 +1,127 @@
...
+   * The text to be displayed inside the button
+   */
+  text: string;
+  /**
+   * The link that the LinkButton should route to.
+   *
+   */
+  href?: string;
+  /**
+   * The target attribute specifies where to open the linked document:
```
Comment (christianvuerings) [inline packages/syntax-core/src/LinkButton/LinkButton.tsx:44]: @somethiiing change the `:` with `.`

File: packages/syntax-core/src/Button/Button.tsx [hunk 18]
```diff
@@ -144,12 +141,16 @@ const Button = forwardRef<HTMLButtonElement, ButtonType>(
           <StartIcon className={classNames(styles.icon, iconSize[size])} />
         )}
         {((loading && loadingText) || (!loading && text)) && (
-          <div className={styles.textContainer}>
```
Comment (christianvuerings) [inline packages/syntax-core/src/Button/Button.tsx:147]: @somethiiing can we remove `textContainer` from the CSS since it's no longer used?

**Review by christianvuerings — APPROVED** [review]
@somethiiing please address the comments before merging

**Pull request merged**